tests/test_scenario/test_link_velocities.py

```diff
@@ -0,0 +1,314 @@
+
+
+def get_random_panda(gazebo: bindings.GazeboSimulator,
+                     world: bindings.World) -> bindings.Model:
+
+    panda_urdf = gym_ignition_models.get_model_file("panda")
+    assert world.insertModel(panda_urdf)
+    assert "panda" in world.modelNames()
+
+    panda = world.getModel("panda")
+
+    joint_space = get_joint_positions_space(model=panda)
+    joint_space.seed(10)
+
+    q = joint_space.sample()
+    dq = joint_space.np_random.uniform(low=-1.0, high=1.0, size=q.shape)
+
+    assert panda.resetJointPositions(q.tolist())
+    assert panda.resetJointVelocities(dq.tolist())
+
+    assert gazebo.run(paused=True)
+    return panda
+
+
+def get_cube(gazebo: bindings.GazeboSimulator,
+             world: bindings.World) -> bindings.Model:
+
+    quaternion = to_wxyz(Rotation.from_euler('x', 45, degrees=True).as_quat())
+    initial_pose = bindings.Pose([0, 0, 0.5], quaternion.tolist())
+
+    cube_urdf = utils.get_cube_urdf()
+    assert world.insertModel(cube_urdf, initial_pose)
+    assert "cube_robot" in world.modelNames()
+
+    cube = world.getModel("cube_robot")
+
+    assert cube.resetBaseWorldLinearVelocity([0.1, -0.2, -0.3])
+    assert cube.resetBaseWorldAngularVelocity([-0.1, 2.0, 0.3])
+
+    assert gazebo.run(paused=True)
+    return cube
+
+
+# 1. VELOCITY LINEAR
+@pytest.mark.parametrize("default_world, get_model, link_name", [
+                          ((1.0 / 10_000, 1.0, 1), get_random_panda, "panda_link7"),
+                          ((1.0 / 10_000, 1.0, 1), get_cube, "cube"),
+                          ],
+                         indirect=["default_world"])
+def test_linear_velocity(
+        default_world: Tuple[bindings.GazeboSimulator, bindings.World],
+        get_model: Callable[[bindings.GazeboSimulator, bindings.World], bindings.Model],
+        link_name: str):
+
+    # Get the simulator and the world
+    gazebo, world = default_world
+    dt = gazebo.stepSize()
+
+    # Get the model
+    model = get_model(gazebo, world)
+
+    # Get the link.
+    # If the link is the base link, get the data through the base methods.
+    link = model.getLink(link_name)
+
+    if link.name() != model.baseFrame():
+
+        position = link.position
+        orientation = link.orientation
+        bodyLinearVelocity = link.bodyLinearVelocity
+        worldLinearVelocity = link.worldLinearVelocity
+
+    else:
+
+        position = model.basePosition
+        orientation = model.baseOrientation
+        bodyLinearVelocity = model.baseBodyLinearVelocity
+        worldLinearVelocity = model.baseWorldLinearVelocity
+
+    # 0.5 seconds of simulation
+    for _ in range(int(0.5 / dt)):
+
+        position_old = np.array(position())
+        assert gazebo.run()
+        position_new = np.array(position())
+
+        world_velocity = (position_new - position_old) / dt
+
+        # Test the world velocity (MIXED)
+        assert world_velocity == pytest.approx(worldLinearVelocity(), abs=1E-2)
```

Comment:
`1E-2` is quite high, what are the velocity values ?

Reply:
The z component of this velocity reaches ~4, x and y are much smaller, less than 1.0.